internal/server/receiver_handlers.go

```diff
@@ -71,8 +87,7 @@ func (s *ReceiverServer) handlePayload() func(w http.ResponseWriter, r *http.Req
 		receivers := make([]apiv1.Receiver, 0)
 		for _, receiver := range allReceivers.Items {
 			if !receiver.Spec.Suspend &&
```

Comment:
Wondering if for suspended resources, we should throw a `http.StatusServiceUnavailable` ([HTTP 503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503)).

Reply:
In addition, I think we expect a webhook path to be unique. Given this, I think we can instruct the `List` call to only return 1 result.

Reply:
See https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Limit

Reply:
about returning a HTTP 503 response for suspended and/or unready resources, i'm not quite sure. according to the mdn docs:
>503 Service Unavailable server error response code indicates that the server is not ready to handle the request.
Common causes are a server that is down for maintenance or that is overloaded.

But a resource not being ready or suspended doesn't really map to the fact that the server is unable to handle the request.

Reply:
maybe we could throw a `http.FailedDependency` ([HTTP 424](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/424)) instead, but according to mdn:
> Regular web servers will normally not return this status code. But some other protocols, like [WebDAV](https://developer.mozilla.org/en-US/docs/Glossary/WebDAV), can return it.

Reply:
> Unavailable server error response code indicates that the server is not ready to handle the request.

It is not ready because "not ready" indicates it isn't (but eventually will be), and "suspend" is usually equal to "maintenance". Which is why I picked this.

Reply:
okay, let's go with 503 since there is no other obvious answer 👍